File: user-service/src/main/java/rs/edu/raf/dto/LoginDto.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.Pattern;

public record LoginDto (
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    String username,
    String password
) { }

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/dto/LoginDTO.kt
package org.example.userservicekotlin.dto

import jakarta.validation.constraints.Pattern

data class LoginDTO(
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    val username: String,
    val password: String
)
